crates/symbolicator-proguard/src/service.rs

```diff
@@ -54,12 +54,18 @@ impl ProguardService {
     /// Retrieves the given [`RemoteFile`] from cache, or fetches it and persists it according
     /// to the provided [`Scope`].
     /// It is possible to avoid using the shared cache using the `use_shared_cache` parameter.
-    pub async fn fetch_file(&self, scope: &Scope, file: RemoteFile) -> CacheEntry<ProguardMapper> {
+    pub async fn fetch_file(
+        &self,
+        scope: &Scope,
+        file: RemoteFile,
+        use_param_mapping: bool,
+    ) -> CacheEntry<ProguardMapper> {
         let cache_key = CacheKey::from_scoped_file(scope, &file);
```

Comment:
TBH, I would just always parse this with support for param mappings.
The way our caching layers work would otherwise mean that you would have to duplicate all the files and create a different `cache_key` for both variants. This is worse in terms of disk usage and memory usage than to just always parse these mappings into memory (and reuse them when needed).

Reply:
Oh shoot, I didn't think about caching at all. Good catch.